### PR TITLE
feat: supply invoke info to pre/post_invoke

### PR DESCRIPTION
**Pull request by pandemicsyn**
In the dbt extension what we do as part of a pre_invoke will actually be influenced by what command will ultimately be invoked. This seems like it might end being somewhat common, and so I think it makes sense to extend the signature for pre_invoke & post_invoke to match that of invoke ...and pass along what name and args will be invoked.

### CODE REVIEW

**Comment by pandemicsyn** [comment]
One other pattern I was tempted to use was doing something like a call back pattern , e.g.

```python
class Somextension(Extensionbase):

   def __init__(self):
        self.register_pre_invoke("some_specific_command", callbackFN)
        self.register_pre_invoke(":splat", othercallBackFN)

   def callbackFN(self):
       # do custom pre_invoke stuff for some_specific_command

   def othercallBackFN(self):
       # do generic stuff for all other commands
```

Originally, I was thinking that a callback pattern might be nice if magically want to configure pre_invoke filters via cookiecutter prompts.

...but that's just as easily doable by pre-populating an if/else in the extensions `pre_invoke`  implementation, AND I think most pre_invoke pattern's will end up being bespoke ones anyway.

File: meltano/edk/extension.py [hunk 1]
```diff
@@ -23,8 +23,13 @@ class DescribeFormat(str, Enum):
 class ExtensionBase(metaclass=ABCMeta):
     """Basic extension interface that must be implemented by all extensions."""
 
-    def pre_invoke(self) -> None:
-        """Called before the extension is invoked."""
+    def pre_invoke(self, invoke_name: str | None, *invoke_args: Any) -> None:
```
Comment (WillDaSilva) [inline meltano/edk/extension.py:26]: @pandemicsyn what is responsible for calling `pre_invoke` and `post_invoke`?

Comment (pandemicsyn) [reply meltano/edk/extension.py:26]: Gets called two ways.

1. [edk.pass_through_invoker](https://github.com/meltano/edk/blob/main/meltano/edk/extension.py#L91 ) which is used by the cookiecutter generated `pass_through_cli` that get's configured for wrappers through which all wrapped calls pass: (https://github.com/meltano/airflow-ext/blob/main/airflow_ext/pass_through.py). 
2. The other spot is from explicit user controlled calls from within custom commands. e.g. the superset extension may call pre_invoke only when "add_user" , or "initialize" are called to ensure that the system db exists.

Comment (WillDaSilva) [reply meltano/edk/extension.py:26]: @pandemicsyn Doesn't `ExtensionBase.pass_through_invoker` need to be updated by this PR? `invoke_name` is not an optional argument.

Comment (pandemicsyn) [reply meltano/edk/extension.py:26]: Doh, yea - you're right.

**Pull request merged**